optaplanner-docs/src/main/asciidoc/Chapter-Partitioned_search/Chapter-Partitioned_search.adoc

```diff
@@ -8,27 +8,214 @@
 
 image::Chapter-Partitioned_search/mapReduceIsTerribleForTsp.png[align="center"]
 
-A future version of Planner will have native support for several forms of partitioning.
-Meanwhile, you can implement it yourself as shown in the image above.
-Use an `Solver` to solve each piece.
+One way to deal with this is to run a non-partitioned Local Search after the Partitioned Search phase.
 
 [NOTE]
 ====
 Not all use cases can be partitioned.
-It only works on use cases for which the planning entities and value ranges can be divided into n pieces, such that none of the constraints cross piece boundaries.
+It only works on use cases for which the planning entities and value ranges can be split into n partition,
+such that none of the constraints cross the boundaries between partitions.
 ====
 
 
+[[partitionedSearchConfiguration]]
+== Configuration
+
+Simplest configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+  </partitionedSearch>
+----
+
+There are several ways to <<partitioningASolution,partition a solution>>.
+
+Advanced configuration:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    ...
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+    <threadFactoryClass>...MyAppServerThreadFactory</threadFactoryClass>
+    <runnablePartThreadLimit>4</runnablePartThreadLimit>
+
+    <constructionHeuristic>...</constructionHeuristic>
+    <localSearch>...</localSearch>
+  </partitionedSearch>
+----
+
+The `threadFactoryClass` allows to plug in a custom `ThreadFactory` for environments
+that don't like arbitrary thread creation,
+such as most application servers, Android or Google App Engine.
+
+The `runnablePartThreadLimit` allows limiting CPU usage, see below.
+
+Just like a `<solver>` element, the `<partitionedSearch>` element can one or more <<solverPhase,phases>>.
+Each of those phases will be run on each partition.
+
+A popular configuration is to first run a Partitioned Search phase
+(which includes a Construction Heuristic and a Local Search)
+and then do a non-partitioned Local Search phase:
+
+[source,xml,options="nowrap"]
+----
+  <partitionedSearch>
+    <solutionPartitionerClass>org.optaplanner.examples.cloudbalancing.variants.solver.partitioner.CloudBalancePartitioner</solutionPartitionerClass>
+
+    <constructionHeuristic/>
+    <localSearch>
+      <secondsSpentLimit>60</secondsSpentLimit>
+    </localSearch>
+  </partitionedSearch>
+  <localSearch/>
+----
+
+
+[[partitioningASolution]]
+== Partitioning a Solution
+
+
+[[customSolutionPartitioner]]
+=== Custom SolutionPartitioner
+
+To use a custom partitioner, configure a partitioner on the Partitioned Search phase:
```

Comment:
"partition" is used three times here and is a little awkward. 

"Use the following example to configure a custom partitioner on the Partitioned Search phase:" ?